src/unitxt/types.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, List, Literal, NewType, TypedDict
+
+from .type_utils import register_type
+
+Text = NewType("Text", str)
+
+
+class Turn(TypedDict):
+    role: Literal["system", "user", "agent"]
+    content: Text
+
+
+Dialog = NewType("Dialog", List[Turn])
+
+
+class Image(TypedDict):
+    image: Any
+
+
+class Audio(TypedDict):
+    audio: Any
+
+
+class Table(TypedDict):
+    header: List[str]
+    rows: List[List[Any]]
```

Comment:
I think that the basic (and common table is a text table, where rows: List[List[Text]]